File: src/java/magma/app/compile/type/resolve/template/FunctionTypes.java
package magma.app.compile.type.resolve.template;

import magma.api.Tuple2;
import magma.api.collect.Joiner;

public final class FunctionTypes {
    public static String generateFunctionType(FunctionType functionType) {
        var joinedArguments = functionType.args()
                .iterWithIndices()
                .map((Tuple2<Integer, String> tuple) -> "arg" + tuple.left() + " : " + tuple.right())
                .collect(new Joiner(", "))
                .orElse("");

        return "(" + joinedArguments + ") => " + functionType.returns();
    }
}

File: src/ts/magma/app/compile/type/template/FunctionTypes.ts
import { FunctionType } from "../../../../../magma/app/compile/type/FunctionType";
import { Tuple2 } from "../../../../../magma/api/Tuple2";
import { Joiner } from "../../../../../magma/api/collect/Joiner";
export class FunctionTypes {
	static generateFunctionType(functionType: FunctionType): string {
		let joinedArguments = functionType.args().iterWithIndices().map((tuple: Tuple2<number, string>) => "arg" + tuple.left() + " : " + tuple.right()/*unknown*/).collect(new Joiner(", ")).orElse("")/*unknown*/;
		return "(" + joinedArguments + ") => " + functionType.returns()/*unknown*/;
	}
}
